src/vivarium/component.py

```diff
@@ -791,7 +798,7 @@ def _register_simulant_initializer(self, builder: Builder) -> None:
 
         if type(self).on_initialize_simulants != Component.on_initialize_simulants:
             builder.population.initializes_simulants(
-                self, creates_columns=self.columns_created, **initialization_requirements
+                self, creates_columns=self.columns_created, **initialization_requirements  # type: ignore[arg-type]
```

Comment:
This block is temporary anyway since we're deprecating old functionalty here. It does highlight the fact, however, that we need to figure out how to best type hint kwargs that get unpacked at the signature. the current solution seems complicated. https://typing.readthedocs.io/en/latest/spec/callables.html#unpack-kwargs